From 2f71e3212afb9834b7910b4a05cf69071569a60c Mon Sep 17 00:00:00 2001
From: Abhishek Arya <inferno@chromium.org>
Date: Sun, 22 Jan 2012 17:47:50 +0800
Subject: [PATCH] Fix some case of pattern parsing errors
To: libvir-list@redhat.com

For https://bugzilla.redhat.com/show_bug.cgi?id=788826
CVE-2011-3970

We could accidentally hit an off by one string array access
due to improper loop exit when parsing patterns

Signed-off-by: Daniel Veillard <veillard@redhat.com>
---
 libxslt/pattern.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/libxslt/pattern.c b/libxslt/pattern.c
index 6161376..1155b54 100644
--- a/libxslt/pattern.c
+++ b/libxslt/pattern.c
@@ -1867,6 +1867,8 @@ xsltCompilePatternInternal(const xmlChar *pattern, xmlDocPtr doc,
 		while ((pattern[end] != 0) && (pattern[end] != '"'))
 		    end++;
 	    }
+	    if (pattern[end] == 0)
+	        break;
 	    end++;
 	}
 	if (current == end) {
-- 
1.7.11.4

